packages/jest/src/index.js

```diff
@@ -1,23 +1,32 @@
-const loaderMerge = require('@neutrinojs/loader-merge');
 const compileLoader = require('@neutrinojs/compile-loader');
 const merge = require('deepmerge');
 const omit = require('lodash.omit');
 const { basename, isAbsolute, join, relative } = require('path');
 const { media, style } = require('neutrino/extensions');
 
 module.exports = (neutrino, options = {}) => {
-  neutrino.config.when(neutrino.config.module.rules.has('lint'), () => {
-    neutrino.use(loaderMerge('lint', 'eslint'), {
-      plugins: ['jest'],
-      envs: ['jest/globals'],
-      rules: {
-        'jest/no-disabled-tests': 'warn',
-        'jest/no-focused-tests': 'error',
-        'jest/no-identical-title': 'error',
-        'jest/valid-expect': 'error'
-      }
-    });
-  });
+  const lintRule = neutrino.config.module.rules.get('lint');
+  if (lintRule) {
+    lintRule.use('eslint').tap(
+      // Don't adjust the lint configuration for projects using their own .eslintrc.
+      lintOptions => lintOptions.useEslintrc
+        ? lintOptions
+        : merge(lintOptions, {
+            baseConfig: {
+              env: {
+                'jest/globals': true
```

Comment:
I've converted the top-level `envs` array form to the object form under `baseConfig.env`, since:
* users will be more familiar with it (ref https://github.com/neutrinojs/neutrino/issues/1166#issuecomment-429067147)
* it then means it's possible to override an `env` by passing eg `'jest/globals': false` (with arrays manual filtering would have to be performed)
* it means the resultant `--inspect` output is much closer to a form that can be copy-pasted into an `.eslintrc.*` file